openex-api/src/main/java/io/openex/rest/settings/SettingsApi.java

```diff
@@ -28,115 +29,139 @@
-                    .map(entry -> {
-                        String uri = "/oauth2/authorization/" + entry.getKey();
-                        String clientName = env.getProperty("openex.provider." + entry.getKey() + ".login");
-                        // In case of missing name configuration, generate a generic name
-                        if (clientName == null) {
-                            clientName = "Login with " + entry.getKey();
-                        }
-                        return new OAuthProvider(entry.getKey(), uri, clientName);
-                    })
-                    .toList();
-        } catch (Exception e) {
-            // No provider defined in the configuration
-            return new ArrayList<>();
-        }
+  private List<OAuthProvider> buildProviders() {
+    try {
+      OAuth2ClientProperties properties = context.getBean(OAuth2ClientProperties.class);
+      Map<String, OAuth2ClientProperties.Registration> providers = properties.getRegistration();
+      return providers.entrySet().stream()
+          .map(entry -> {
+            String uri = "/oauth2/authorization/" + entry.getKey();
+            String clientName = env.getProperty("openex.provider." + entry.getKey() + ".login");
+            // In case of missing name configuration, generate a generic name
+            if (clientName == null) {
+              clientName = "Login with " + entry.getKey();
+            }
+            return new OAuthProvider(entry.getKey(), uri, clientName);
+          })
+          .toList();
+    } catch (Exception e) {
+      // No provider defined in the configuration
+      return new ArrayList<>();
     }
+  }
 
-    private PlatformSetting buildPlatformSetting(Map<String, Setting> dbSettings, SETTING_KEYS setting) {
-        Optional<Setting> platformName = ofNullable(dbSettings.get(setting.key()));
-        String settingValue = platformName.map(Setting::getValue).orElse(setting.defaultValue());
-        return new PlatformSetting(setting.key(), settingValue);
+  private List<OAuthProvider> buildSaml2Providers() {
+    try {
+      Saml2RelyingPartyProperties properties = context.getBean(Saml2RelyingPartyProperties.class);
+      System.out.println(properties);
+      Map<String, Saml2RelyingPartyProperties.Registration> providers = properties.getRegistration();
+      return providers.entrySet().stream()
+          .map(entry -> {
+            String uri = "/saml2/authenticate/" + entry.getKey();
+            String clientName = env.getProperty("openex.provider." + entry.getKey() + ".login");
+            // In case of missing name configuration, generate a generic name
+            if (clientName == null) {
+              clientName = "Login with " + entry.getKey();
+            }
+            return new OAuthProvider(entry.getKey(), uri, clientName);
+          })
+          .toList();
+    } catch (Exception e) {
+      // No provider defined in the configuration
+      return new ArrayList<>();
     }
```

Comment:
We can have one buildProviders with line 89 to 103
Rename the actuel buildProviders to buildOauth2Providers
And call  buildProviders method inside buildOauth2Providers and buildSaml2Providers